### ISSUE TITLE
Migrating to .Net Core / .NET v7.0 😎

### ISSUE THREAD

**Issue by w8tcha** [YAFNET/YAFNET#814]
I post here the migration process from ASP.NET Webforms (.NET Framework 4.8) to ASP.NET Core (.NET v7.0).

The current plan is to put the current YAF Version (3.1.x) in to the netfx Branch. This version will be continue to exist but will receive only bug fixes. No more new features.

The main master branch will soon have the next version of YAF (v4.0.0) for .NET 7.

At the moment Version 4 is build with Razor Pages, but its possible that in the future there could also be a version with Blazor.

I try to commit the first version in the next 2 weeks. So feel free to post questions about that topic here.

**Comment by KevinDuganJr** [comment]
I think this will be quite interesting to see the transition. 

Aside from moving away from legacy technology (e.g. Web Forms) and going to a more modern technology (e.g. Razor Pages), what benefits do you envision for YAF?

**Comment by w8tcha** [comment]
For me the main benefit and reason to migrate is cost and security. For me windows server hosting is too expensive and insecure. Linux is so much better in that regard.

**Comment by KevinDuganJr** [comment]
> For me the main benefit and reason to migrate is cost and security. For me windows server hosting is too expensive and insecure. Linux is so much better in that regard.

That is true, Windows does cost more. I'm paying $5 a month for one windows host and I think my other one is $7.99 a month. I'm in the middle of transitioning my sites from one host to the other.

**Comment by w8tcha** [comment]
I was referring to server hosting not site hosting. At the moment its about 300€. Plus the cost for the windows server and sql server licenses.

**Comment by KevinDuganJr** [comment]
> I was referring to server hosting not site hosting. At the moment its about 300€. Plus the cost for the windows server and sql server licenses.

Oh, wow, that is super expensive. I thought maybe windows shared hosting was just a lot more. I didn't realize you needed an entire server. I only host a handful of websites and none of them bring in any donations or revenue, so it's all just out of pocket for me. 

I debated on going with a Virtual Private Server, but that is about $30 a month, which includes MSSQL 2019 and Windows Server 2022, but $30 a month for just handful of websites -- nah, haha. I more or less considered it because I wanted to get some experience administrating a server and maybe run some custom tasks in the background instead of on a computer at home.

**Comment by thecarnie** [comment]
> I was referring to server hosting not site hosting. At the moment its about 300€. Plus the cost for the windows server and sql server licenses.

I wish I would have paid more attention on the forums - I might have been able to assist you earlier on hosting/pricing, etc.

We run multiple YAF forums on a server that just runs SQL Express (free) edition and it handles the traffic just fine - for what it's worth.  You would have to be servicing thousands of YAF users an hour before you really overload the Express edition.

That said, you should be able to at least reduce your hosting costs if you're paying for a higher version of SQL Server --

Still looking forward to the updated version regardless!

**Comment by w8tcha** [comment]
I use Sql express, but i consider switching to mariadb

**Comment by ferib** [comment]
Can we somewhere see the progress being made for porting .net 7.0?

**Comment by w8tcha** [comment]
I commited the first beta. The master branch now contains the YAF 4.0 BETA and the netfx branch the current stable 3.1.x release

**Comment by Drizzle365** [comment]
When will the first version 4.0 be released.
Looking forward to it

**Comment by w8tcha** [comment]
It still needs some testing, i will update the test and support forum soon. But i guess the release will be before .net 8 will be released in November/Dezember.

Maybe it would be good to release YAF 4.0 in september 20 Years after the first release.

**Comment by brose1977** [comment]
Will the .NET Core version of the site support embedding it as a control?

**Comment by w8tcha** [comment]
> Will the .NET Core version of the site support embedding it as a control?

Asp.net core does not have the concept of user Controls, that means it's not possible to provide a similar solution like in webforms to embed the forum as control.

However its still possible to embed the forum in your asp.net core application. You could run the forum as sub application, and then modify the _Layout.cshtml to match the main application

**Comment by brose1977** [comment]
Thanks for the quick response.  I am actually returning to .NET after a long hiatus (.NET 4.5).  I've looked around for the concept of a sub application but am only seeing things that are related to IIS.  I plan on using the Kestrel server.  What do you mean exactly when you say "sub-application"?

**Comment by w8tcha** [comment]
I was referring to the iis solution for sub applications

https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/advanced?view=aspnetcore-7.0&source=recommendations#sub-applications

Be aware for me this is all new too. Everything which is not iis I can only try to Google or read the ms documentations.

**Comment by w8tcha** [comment]
I uploaded the YAF 4.0.0-beta.1 packages

https://github.com/YAFNET/YAFNET/releases/tag/v4.0.0-beta.1

**Comment by KevinDuganJr** [comment]
This is kind of neat. I might try upgrading my test site to this.

**Comment by ElderJames** [comment]
Hello @w8tcha , do you plan to use Blazor as a webui? I recommend the open source [antblazor ](https://github.com/ant-design-blazor/ant-design-blazor) for such a great forum system.

**Comment by w8tcha** [comment]
No, maybe as an alternative solution. But you blazor library in incompatible. There are much better solutions out there.

**Comment by ElderJames** [comment]
That's fine. Which solution do you prefer?

**Comment by digitarenet** [comment]
There is a release date for .NET Core Stable version (approximately)?
